### PR TITLE
fix: dont resolve endpoint url

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

Ref https://github.com/eggjs/egg/issues/777

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
+    "egg": "^1.1.0",
+    "egg-bin": "^3.3.0",
+    "egg-mock": "^3.4.0",
+    "eslint": "^3.19.0",
     "eslint-config-egg": "^3.2.0",
     "is-type-of": "^1.0.0",
     "pedding": "^1.1.0",
-    "supertest": "^3.0.0",
-    "urllib": "2"
+    "supertest": "^3.0.0"
```
Comment (atian25) [inline package.json]: 用 `app.httpRequest` ?

File: package.json [hunk 2]
```diff
@@ -42,7 +42,8 @@
     "test": "npm run lint -- --fix && npm run test-local",
     "test-local": "egg-bin test",
     "cov": "egg-bin cov",
-    "ci": "npm run lint && npm run cov"
+    "pkgfiles": "egg-bin pkgfiles",
+    "ci": "egg-bin pkgfiles --check && npm run lint && npm run cov"
```
Comment (atian25) [inline package.json]: 这句看来可以加到 boilerplate 去

File: package.json (hunk 1 above)
Comment (popomore) [inline package.json]: 我改下

**Comment by atian25** [comment]
ci 挂了，要加下 `config.keys`

File: lib/oss.js [hunk 3]
```diff
@@ -3,16 +3,11 @@
...
 const OSS = require('ali-oss');
 
-const RE_HTTP_PROTOCOL = /^https?:\/\//;
 function checkBucketConfig(config) {
   assert(config.endpoint || config.region,
     '[egg-oss] Must set `endpoint` or `region` in oss\'s config');
   assert(config.accessKeySecret && config.accessKeyId,
     '[egg-oss] Must set `accessKeyId` and `accessKeySecret` in oss\'s config');
-
-  if (config.endpoint && RE_HTTP_PROTOCOL.test(config.endpoint)) {
```
Comment (popomore) [inline lib/oss.js]: 这个逻辑有问题，我去掉了

https://github.com/ali-sdk/ali-oss/blob/master/lib/client.js#L490

**Comment by fengmk2** [comment]
1.0.2

**Pull request merged**